spec/data-types-and-values.html

```diff
@@ -159,8 +159,24 @@ <h1>
 
     <emu-clause id="sec-ecmascript-language-types-box-type">
       <h1>The Box Type</h1>
-      <p>The Box type is the set of all the possible singleton primitive wrappers around any ECMAScript value. Each box value holds an associated [[Value]] containing an ECMAScript value. The [[Value]] internal slot is never modified.</p>
+      <p>The Box type is the set of all the possible singleton primitive wrappers around any ECMAScript value. Each box value holds an associated [[Value]] containing an ECMAScript value and a [[Realm]] containing the Realm of the Box function which created the Box value. The [[Value]] and [[Realm]] internal slots are never modified.</p>
 
+      <emu-clause id="sec-ecmascript-language-types-box-unbox" type="abstract operation">
+        <h1>
+          Unbox (
+            _box_: a Box,
+          )
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It extracts the boxed Object from _box_, throwing if it has been boxed by a different Realm's Box constructor.</dd>
+        </dl>
+        <emu-alg>
+          1. Let _realm_ be the current Realm Record.
+          1. If _box_.[[Realm]] is not _realm_, throw a TypeError exception.

```

Comment:
This would make it completely untenable for membranes, as there's be no way to remember the mapping of a record/tuple which only contains boxes of primitives.

Reply:
Q: Why does it need to be remembered? Can't membranes rebuild the box every time, since it doesn't have identity?

Reply:
That means every time you send a record/tuple through the membrane that `Box.containsBox` but does not `Object.containsObject`, you have to recursively unbox all the boxes, push their content, and remap the record/tuple/box on the other side, without being able to remember this work. That would be incredibly inefficient.